backend/qto_producer.py

```
@@ -769,3 +777,101 @@ def batch_upsert_manual_elements(self, project_id: ObjectId, elements_data: List
+
+    def get_ifc_processing_job(self, job_id_str: str) -> Optional[Dict[str, Any]]:
+        """Retrieves an IFC processing job by its string ID."""
+        if self.db is None:
+            logger.error(f"MongoDB not connected, cannot get job {job_id_str}")
+            return None
+        try:
+            job_oid = ObjectId(job_id_str)
+            job_doc = self.db.ifc_processing_jobs.find_one({"_id": job_oid})
+            if job_doc:
+                 # Ensure _id is returned as a string if present, for Pydantic model compatibility
+                if "_id" in job_doc:
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)